chargehound/chargehound-node
# DTFPCHG-178: Fixing snyk issue

Fixing snyk issue
Modified the fee display UI. Updated the stats logic for Stripe fee reversal

## JIRA ID | Git Issue #
[DTFPCHG-178](https://paypal.atlassian.net/browse/DTFPCHG-178)

## Change Type
- [X] Bug fix (is this a backward compatible change?)
- [ ] New feature (is this a backward compatible change?)
- [ ] Breaking change (is this a non-backward compatible change?)
- [ ] DB changes (add scripts in the sql/migrations/{date} folder)

## Change Impact
####  Summary
- The SAST tool [Snyk] has identified security risks with code snippet in this repository.
- http.request method is being used in lib/api-requestor.js file which is the root cause for this issue. 
- We are mainly using this for local testing since we do not support http in the Prod environment. 
- Removed the support for making the request using http protocol.
- At any point in time if we want to continue testing in our local, we can go ahead and add this code snippet. 

### Demo
![image](https://github.com/chargehound/chargehound-node/assets/120496064/f26e7038-624b-4224-b7aa-bdb9549d9953)
![image](https://github.com/chargehound/chargehound-node/assets/120496064/a624e1b2-9ecd-454a-a4ca-e64d3d4b511b)
![image](https://github.com/chargehound/chargehound-node/assets/120496064/861fb425-9072-4796-b7de-00d5da2c7ad6)

### Notes
- Updated the lib/api-requestor.js file to remove the support for http protocol. 

## Testing Instructions

 * Try to submit a dispute in our local environment using http protocol. Configure the required properties in the lib/index.js 
 * Observe that we would be able to submit the test dispute as long as http protocol is supported in our SDK. 
 * Now go ahead and remove the http protocol support from lib/api-requestor.js file and try submitting a test dispute. 
 * Observe that the request would fail with the above mentioned error (in the screenshot).
